File: create_request.go
// Random Go struct generated on 2025-07-15
package main

import (
    "encoding/json"
    "fmt"
    "time"
)

type GetResponse struct {
    Name      string    `json:"name"`
    Value     int       `json:"value"`
    Active    bool      `json:"active"`
    CreatedAt time.Time `json:"created_at"`
}

func NewGetResponse(name string, value int) *GetResponse {
    return &GetResponse{
        Name:      name,
        Value:     value,
        Active:    true,
        CreatedAt: time.Now(),
    }
}

func (s *GetResponse) UpdateValue(newValue int) {
    s.Value = newValue
}

func (s *GetResponse) ToJSON() ([]byte, error) {
    return json.Marshal(s)
}

func main() {
    obj := NewGetResponse("sample", 30)
    data, _ := obj.ToJSON()
    fmt.Println(string(data))
}

File: create_request.c
// Random C struct generated on 2025-08-27
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char name[100];
    int value;
    int active;
    long timestamp;
} HandleRequest;

HandleRequest* create_handlerequest(const char* name, int value) {
    HandleRequest* obj = malloc(sizeof(HandleRequest));
    if (obj == NULL) return NULL;
    
    strncpy(obj->name, name, sizeof(obj->name) - 1);
    obj->name[sizeof(obj->name) - 1] = '\0';
    obj->value = value;
    obj->active = 1;
    obj->timestamp = 1644660562;
    
    return obj;
}

void update_handlerequest_value(HandleRequest* obj, int new_value) {
    if (obj != NULL) {
        obj->value = new_value;
    }
}

void print_handlerequest(const HandleRequest* obj) {
    if (obj != NULL) {
        printf("Name: %s, Value: %d, Active: %d\n", 
               obj->name, obj->value, obj->active);
    }
}

void free_handlerequest(HandleRequest* obj) {
    if (obj != NULL) {
        free(obj);
    }
}

int main() {
    HandleRequest* obj = create_handlerequest("sample", 21);
    print_handlerequest(obj);
    free_handlerequest(obj);
    
    return 0;
}
